gorb/gorb.go

```diff
@@ -0,0 +1,323 @@
+/*
+Package gorb provide the ability to register and deregister the backend
+*/
+package gorb
+
+import (
+	"errors"
+
+	"time"
+
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"io"
+
+	"path"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/hashicorp/go-multierror"
+	"github.com/sky-uk/feed/controller"
+)
+
+// HealthcheckArgs defines health check URI
+type HealthcheckArgs struct {
+	Method string `json:"method"`
+	Path   string `json:"path"`
+	Expect string `json:"expect"`
+}
+
+// Pulse defines backend health check
+type Pulse struct {
+	TypeHealthcheck string          `json:"type"`
+	Args            HealthcheckArgs `json:"args"`
+	Interval        string          `json:"interval"`
+}
+
+// BackendConfig defines the backend configuration
+type BackendConfig struct {
+	Host   string `json:"host"`
+	Port   int    `json:"port"`
+	Method string `json:"method"`
+	Weight int    `json:"weight"`
+	Pulse  Pulse  `json:"pulse"`
+}
+
+// Backend defines the backend configuration + service name
+type backend struct {
+	ServiceName   string
+	BackendConfig BackendConfig
+}
+
+type loopbackAction string
+
+const (
+	addLoopback    loopbackAction = "add"
+	deleteLoopback loopbackAction = "del"
+)
+
+// New creates a gorb handler
+func New(serverBaseURL string, instanceIP string, drainDelay time.Duration, servicesDefinition string, backendWeight int, backendMethod string, vipLoadbalancer string, manageLoopback bool, backendHealthcheckInterval string, interfaceProcFsPath string) (controller.Updater, error) {
+	if serverBaseURL == "" {
+		return nil, errors.New("unable to create Gorb updater: missing server ip address")
+	}
+	initMetrics()
+	log.Infof("Gorb server url: %s, drainDelay: %v, instance ip adddress: %s, vipLoadbalancer: %s ", serverBaseURL, drainDelay, instanceIP, vipLoadbalancer)
+
+	backendDefinitions := []backend{}
+	servicesDefinitionArr := strings.Split(servicesDefinition, ",")
+
+	var backendDefinition backend
+	for _, service := range servicesDefinitionArr {
+
+		servicesArr := strings.Split(service, ":")
+		port, err := strconv.Atoi(servicesArr[1])
```

Comment:
I think should be doing the string parsing in main.go, where we can log fatal error or whatever at startup, and our library should just take in some type with the fields set. This can be done pretty easily using a custom flag type.

Reply:
we'll do this in the main, it's best to validate the arguments there

Reply:
I like the idea of the custom flag type, but it's lacking the default value which I think is useful in this case to output the the correct format on the command line.